src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -390,10 +390,14 @@ protected sealed override void LazyAsyncMethodChecks(CancellationToken cancellat
 
             if (!this.IsAsync)
             {
-                state.NotePartComplete(CompletionPart.StartAsyncMethodChecks);
-                if (state.NotePartComplete(CompletionPart.FinishAsyncMethodChecks) && IsPartialDefinition)
+                if (state.NotePartComplete(CompletionPart.StartAsyncMethodChecks))
                 {
-                    DeclaringCompilation.SymbolDeclaredEvent(this);
+                    if (IsPartialDefinition) DeclaringCompilation.SymbolDeclaredEvent(this);
```

Comment:
It would be good to have a comment explaining why only partial definitions are handled here. Where do we raise event for a non-partial method.